provider/spark.go

```diff
@@ -1230,44 +1278,59 @@ func (spark *SparkOfflineStore) sqlTransformation(config TransformationConfig, i
-	transformationFile, err := spark.Store.NewestFileOfType(spark.Store.PathWithPrefix(ResourcePrefix(config.TargetTableID), false), Parquet)
+	// TODO: understand why the trailing slash is needed
+	// transformationDestinationWithSlash := strings.Join([]string{transformationDestination.PathWithBucket(), ""}, "/")
+	// spark.Logger.Infow("Transformation Destination With Slash", "dest", transformationDestinationWithSlash)
+
+	transformationDirPath, err := spark.Store.CreateDirPath(GetTransformationFileLocation(config.TargetTableID))
+	if err != nil {
+		return fmt.Errorf("could not create directory path for spark transformation: %v", err)
+	}
+
+	transformationFilepath, err := spark.Store.NewestFileOfType(transformationDirPath, filestore.Parquet)
 	if err != nil {
 		return fmt.Errorf("error checking if transformation file exists")
 	}
-	spark.Logger.Infow("Transformation file", "dest", transformationFile)
-	transformationExists := transformationFile != ""
+	spark.Logger.Infow("Transformation file", "dest", transformationFilepath.PathWithBucket())
+	transformationExists := transformationFilepath.Key() != ""
 	if !isUpdate && transformationExists {
-		spark.Logger.Errorw("Transformation already exists", config.TargetTableID, transformationDestination)
-		return fmt.Errorf("transformation %v already exists at %s", config.TargetTableID, transformationDestination)
+		spark.Logger.Errorw("Transformation already exists", config.TargetTableID, transformationDestination.PathWithBucket())
+		return fmt.Errorf("transformation %v already exists at %s", config.TargetTableID, transformationDestination.PathWithBucket())
 	} else if isUpdate && !transformationExists {
-		spark.Logger.Errorw("Transformation doesn't exists at destination and you are trying to update", config.TargetTableID, transformationDestination)
-		return fmt.Errorf("transformation %v doesn't exist at %s and you are trying to update", config.TargetTableID, transformationDestination)
+		spark.Logger.Errorw("Transformation doesn't exists at destination and you are trying to update", config.TargetTableID, transformationDestination.PathWithBucket())
+		return fmt.Errorf("transformation %v doesn't exist at %s and you are trying to update", config.TargetTableID, transformationDestination.PathWithBucket())
 	}
 
-	transformationFilePath := spark.Store.PathWithPrefix(GetTransformationFileLocation(config.TargetTableID), false)
-	spark.Logger.Infow("Transformation file path", "dest", transformationFilePath)
-	fileName := "transformation.pkl"
-	transformationFileLocation := fmt.Sprintf("%s/%s", transformationFilePath, fileName)
-	spark.Logger.Infow("Uploading Transformation File", "location", transformationFileLocation)
-	if err := spark.Store.Write(transformationFileLocation, config.Code); err != nil {
+	pklFilepath, err := spark.Store.CreateFilePath(fmt.Sprintf("featureform/DFTransformations/%s/%s/transformation.pkl", config.TargetTableID.Name, config.TargetTableID.Variant))
```

Comment:
this should be in a function somewhere

Reply:
Agreed we're still creating paths with Sprintf